pkg/testutils/cluster.go

```diff
@@ -0,0 +1,202 @@
+// +build functional
+
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), as published by the Free Software Foundation,
+// or under the Apache License, Version 2.0 <LICENSE-APACHE2 or
+// http://www.apache.org/licenses/LICENSE-2.0>.
+//
+// You may not use this file except in compliance with those terms.
+//
+
+package testutils
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	client "github.com/heketi/heketi/client/api/go-client"
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/heketi/heketi/pkg/utils"
+
+	"github.com/heketi/tests"
+)
+
+type ClusterEnv struct {
+	Nodes   []string
+	SSHPort string
+	Disks   []string
+
+	HeketiUrl string
+}
+
+func (ce *ClusterEnv) Update() {
+	// update the port from the env
+	env := os.Getenv("HEKETI_TEST_STORAGEPORT")
+	if "" != env {
+		ce.SSHPort = env
+	}
+
+	// update the node IPs from the env
+	for i := 0; i < len(ce.Nodes); i++ {
+		key := fmt.Sprintf("HEKETI_TEST_STORAGE%v", i)
+		value := os.Getenv(key)
+		if "" != value {
+			ce.Nodes[i] = value
+		}
+	}
+}
+
+func (ce *ClusterEnv) client() *client.Client {
+	return client.NewClientNoAuth(ce.HeketiUrl)
+}
+
+func (ce *ClusterEnv) SshHost(index int) string {
```

Comment:
As this is an Exposed function can we have a check for the slice length to avoid `index out of range error`?

Reply:
Because this is meant for test code and I'm OK with panics in test code I think I'd like to leave it as-is for now to retain a simpler interface.

Reply:
good point, @Madhu-1 , but I'm ok with John's justification.